app/src/main/java/com/example/mymusic/ui/base/BaseActivity.kt

```diff
@@ -0,0 +1,72 @@
+package com.example.mymusic.ui.base
+
+import android.annotation.TargetApi
+import android.app.ActivityManager
+import android.app.ProgressDialog
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
+import android.view.inputmethod.InputMethodManager
+import androidx.appcompat.app.AppCompatActivity
+import com.example.mymusic.utils.CommonUtils
+
+
+abstract class BaseActivity : AppCompatActivity(), MvpView, BaseFragment.Callback {
+    private var mProgressDialog: ProgressDialog? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+    }
+
+    @TargetApi(Build.VERSION_CODES.M)
+    fun requestPermissionsSafely(permissions: Array<String?>?, requestCode: Int) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            requestPermissions(permissions!!, requestCode)
+        }
+    }
+
+    @TargetApi(Build.VERSION_CODES.M)
+    fun hasPermission(permission: String?): Boolean {
+        return Build.VERSION.SDK_INT < Build.VERSION_CODES.M ||
+                checkSelfPermission(permission!!) == PackageManager.PERMISSION_GRANTED
+    }
+
+    override fun showLoading() {
+        hideLoading()
+        mProgressDialog = CommonUtils.showLoadingDialog(this)
+        mProgressDialog!!.show()
```

Comment:
Không sử dụng !! nhé, như này nó vẫn bỏ qua check null và vẫn throws NullPointerException nếu null 
```suggestion
        mProgressDialog?.show()
```